vllm/core/block/cpu_gpu_block_allocator.py

```diff
@@ -105,11 +105,24 @@ def __init__(
             Device.GPU: gpu_block_allocator,
         }
 
+        self._null_block: Optional[Block] = None
+
         self._block_ids_to_allocator: Dict[int, BlockAllocator] = {}
         for _, allocator in self._allocators.items():
             for block_id in allocator.all_block_ids:
                 self._block_ids_to_allocator[block_id] = allocator
 
+    @property
+    def null_block(self) -> Block:
+        if self._null_block is None:
+            self._null_block = self.allocate_mutable(None, Device.GPU)
```

Comment:
Not super familiar with interface, but if we use allocate_immutable, isn't this supposed to guarantee the hack below?

Reply:
It doesn't seem to prevent modifications, at least for the `NaiveBlocks` - we now use a `NullBlock` wrapper anyways. `allocate_immutable()` makes the block participate in hash tables of valid prefixes. I'm not sure we want the null block there (and even then, it's probably not very useful), so I kept `allocate_mutable()`